lib/child_process.js

```diff
@@ -18,7 +18,13 @@ const setupChannel = child_process.setupChannel;
 const ChildProcess = exports.ChildProcess = child_process.ChildProcess;
 
 function stdioStringToArray(option) {
-  return [option, option, option, 'ipc'];
+  switch (option) {
+    case 'ignore':
+    case 'pipe':
+    case 'inherit':
+      return [option, option, option, 'ipc'];
+  }
+  throw new TypeError('Incorrect value of stdio option: ' + option);
```

Comment:
I think this should instead go inside the `default:` case in the `switch` above.